bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/builder/ThingStatusInfoBuilder.java

```diff
@@ -82,8 +84,7 @@ private ThingStatusInfoBuilder(ThingStatus status, ThingStatusDetail statusDetai
      *
      * @throws IllegalArgumentException if thing status detail is null
      */
-    public @NonNull ThingStatusInfoBuilder withStatusDetail(@NonNull ThingStatusDetail statusDetail)
-            throws IllegalArgumentException {
+    public ThingStatusInfoBuilder withStatusDetail(ThingStatusDetail statusDetail) throws IllegalArgumentException {
```

Comment:
Not related to your changes, but do you happen to know why there is a `throws IllegalArgumentException`?

Reply:
Good catch. These were leftovers from before our `null` annotations, i.e. from before the first `null` annotations were added to the file. When they were introduced the `throw IAE` in the body of the methods was removed but not in its header. I have corrected it in this PR now.